src/wp-includes/class-wp-application-passwords.php

```diff
@@ -0,0 +1,400 @@
+		add_filter( 'determine_current_user', array( __CLASS__, 'rest_api_auth_handler' ), 20 );
+		self::fallback_populate_username_password();
+	}
+
+	/**
+	 * Handle declaration of REST API endpoints.
+	 *
+	 * @since ?.?.0
+	 *
+	 * @access public
+	 * @static
+	 */
+	public static function rest_api_init() {
+		// Some hosts that run PHP in FastCGI mode won't be given the Authentication header.
+		register_rest_route(
+			'2fa/v1',
+			'/test-basic-authorization-header/',
+			array(
+				'methods'             => WP_REST_Server::READABLE . ', ' . WP_REST_Server::CREATABLE,
+				'callback'            => __CLASS__ . '::rest_test_basic_authorization_header',
+				'permission_callback' => '__return_true',
+			)
+		);
+	}
+
+	/**
+	 * Loosely Based on https://github.com/WP-API/Basic-Auth/blob/master/basic-auth.php
+	 *
+	 * @since ?.?.0
+	 *
+	 * @access public
+	 * @static
+	 *
+	 * @param $input_user
+	 *
+	 * @return WP_User|bool
+	 */
+	public static function rest_api_auth_handler( $input_user ) {
+		// Don't authenticate twice.
+		if ( ! empty( $input_user ) ) {
+			return $input_user;
+		}
+
+		// Check that we're trying to authenticate
+		if ( ! isset( $_SERVER['PHP_AUTH_USER'] ) ) {
+			return $input_user;
+		}
+
+		$user = self::authenticate( $input_user, $_SERVER['PHP_AUTH_USER'], $_SERVER['PHP_AUTH_PW'] );
+
+		if ( $user instanceof WP_User ) {
+			return $user->ID;
+		}
+
+		// If it wasn't a user what got returned, just pass on what we had received originally.
+		return $input_user;
+	}
+
+	/**
+	 * Test whether PHP can see Basic Authorization headers passed to the web server.
+	 *
+	 * @return WP_Error|array
+	 */
+	public static function rest_test_basic_authorization_header() {
+		$response = array();
+
+		if ( isset( $_SERVER['PHP_AUTH_USER'] ) ) {
+			$response['PHP_AUTH_USER'] = $_SERVER['PHP_AUTH_USER'];
+		}
+
+		if ( isset( $_SERVER['PHP_AUTH_PW'] ) ) {
+			$response['PHP_AUTH_PW'] = $_SERVER['PHP_AUTH_PW'];
+		}
+
+		if ( empty( $response ) ) {
+			return new WP_Error( 'no-credentials', __( 'No HTTP Basic Authorization credentials were found submitted with this request.' ), array( 'status' => 404 ) );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Some servers running in CGI or FastCGI mode don't pass the Authorization
+	 * header on to WordPress.  If it's been rewritten to the `REMOTE_USER` header,
+	 * fill in the proper $_SERVER variables instead.
+	 */
+	public static function fallback_populate_username_password() {
+		// If we don't have anything to pull from, return early.
+		if ( ! isset( $_SERVER['REMOTE_USER'] ) && ! isset( $_SERVER['REDIRECT_REMOTE_USER'] ) ) {
+			return;
+		}
+
+		// If either PHP_AUTH key is already set, do nothing.
+		if ( isset( $_SERVER['PHP_AUTH_USER'] ) || isset( $_SERVER['PHP_AUTH_PW'] ) ) {
+			return;
+		}
+
+		// From our prior conditional, one of these must be set.
+		$header = isset( $_SERVER['REMOTE_USER'] ) ? $_SERVER['REMOTE_USER'] : $_SERVER['REDIRECT_REMOTE_USER'];
```

Comment:
```suggestion
		$header = isset( $_SERVER['HTTP_AUTHORIZATION'] ) ? $_SERVER['HTTP_AUTHORIZATION'] : $_SERVER['REDIRECT_HTTP_AUTHORIZATION'];
```
Can we use `HTTP_AUTHORIZATION` instead to match https://core.trac.wordpress.org/ticket/47077

Reply:
Implemented.